internal/events/handler.go

```diff
@@ -96,74 +88,17 @@ func (h *EventHandlerImpl) HandleEventBatch(ctx context.Context, batch EventBatc
 }
 
 func (h *EventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Configuration) error {
-	// Write all secrets (nuke and pave).
-	// This will remove all secrets in the secrets directory before writing the requested secrets.
-	// FIXME(kate-osborn): We may want to rethink this approach in the future and write and remove secrets individually.
```

Comment:
now, in file.Manager, we remove all files, including Secrets and replace them. 
Let me know if we want to have a FIXME for that or we can update [the](https://github.com/nginxinc/nginx-kubernetes-gateway/issues/561)

Reply:
https://github.com/nginxinc/nginx-kubernetes-gateway/issues/561